src/pybind/mgr/cephadm/schedule.py

```diff
@@ -353,14 +353,7 @@ def expand_candidates(ls: List[DaemonPlacement], num: int) -> List[DaemonPlaceme
             for i in range(len(to_add)):
                 to_add[i] = to_add[i].assign_rank_generation(ranks[i], self.rank_map)
 
-        # If we don't have <count> the list of candidates is definitive.
-        if count is None:
-            final = existing_slots + to_add
-            logger.debug('Provided hosts: %s' % final)
-            return self.place_per_host_daemons(final, to_add, to_remove)
```

Comment:
why is this being removed? We don't need static hashing for situations where the placement is explicit so this code block here is correct. Swapping the order of the hosts list we return when which hosts are actually going to be in the list is certain is irrelevant as we'll just place the daemons on each host in the list either way.